source/opt/dead_branch_elim_pass.cpp

```diff
@@ -73,260 +72,215 @@ bool DeadBranchElimPass::GetConstInteger(uint32_t selId, uint32_t* selVal) {
-      // with lidx.
-      uint32_t lcnt = 0;
-      uint32_t lidx = 0;
-      uint32_t icnt = 0;
-      inst->ForEachInId([&deadPreds, &icnt, &lcnt, &lidx, this](uint32_t* idp) {
-        if (icnt % 2 == 1) {
-          if (deadPreds.find(cfg()->block(*idp)) == deadPreds.end()) {
-            ++lcnt;
-            lidx = icnt - 1;
+        if (changed) {
+          // Replace the phi with either a single value or a rebuilt phi.
+          uint32_t replId;
+          if (operands.size() == 2) {
+            replId = operands[0].words[0];
+          } else {
+            replId = TakeNextId();
+            std::unique_ptr<ir::Instruction> newPhi(
+                new ir::Instruction(context(), SpvOpPhi, inst->type_id(),
+                                    replId, std::move(operands)));
+            get_def_use_mgr()->AnalyzeInstDefUse(&*newPhi);
+            context()->set_instr_block(&*newPhi, &block);
+            inst->InsertBefore(std::move(newPhi));
           }
+          context()->ReplaceAllUsesWith(inst->result_id(), replId);
+          iter = context()->KillInst(&*inst);
+        } else {
+          ++iter;
         }
-        ++icnt;
-      });
-      // If just one live predecessor, replace resultid with live value id.
-      uint32_t replId;
-      if (lcnt == 1) {
-        replId = inst->GetSingleWordInOperand(lidx);
-      } else {
-        // Otherwise create new phi eliminating dead predecessor entries
-        assert(lcnt > 1);
-        replId = TakeNextId();
-        std::vector<ir::Operand> phi_in_opnds;
-        icnt = 0;
-        uint32_t lastId;
-        inst->ForEachInId(
-            [&deadPreds, &icnt, &phi_in_opnds, &lastId, this](uint32_t* idp) {
-              if (icnt % 2 == 1) {
-                if (deadPreds.find(cfg()->block(*idp)) == deadPreds.end()) {
-                  phi_in_opnds.push_back(
-                      {spv_operand_type_t::SPV_OPERAND_TYPE_ID, {lastId}});
-                  phi_in_opnds.push_back(
-                      {spv_operand_type_t::SPV_OPERAND_TYPE_ID, {*idp}});
-                }
-              } else {
-                lastId = *idp;
-              }
-              ++icnt;
-            });
-        std::unique_ptr<ir::Instruction> newPhi(new ir::Instruction(
-            context(), SpvOpPhi, inst->type_id(), replId, phi_in_opnds));
-        get_def_use_mgr()->AnalyzeInstDefUse(&*newPhi);
-        inst->InsertBefore(std::move(newPhi));
       }
-      const uint32_t phiId = inst->result_id();
-      context()->KillNamesAndDecorates(phiId);
-      (void)context()->ReplaceAllUsesWith(phiId, replId);
-      inst = context()->KillInst(inst);
     }
   }
 
-  // Erase dead blocks
-  for (auto ebi = func->begin(); ebi != func->end();)
-    if (elimBlocks.find(&*ebi) != elimBlocks.end())
+  // Erase dead blocks.
+  for (auto ebi = func->begin(); ebi != func->end();) {
+    if (unreachableMerges.count(&*ebi)) {
+      // Make unreachable, but leave the label.
+      KillAllInsts(&*ebi, false);
```

Comment:
It is hard to know what the false means here, even with the comment above.  Consider the suggestion inhttps://google.github.io/styleguide/cppguide.html#Implementation_Comments, "Function Argument Comments" section.